src/main/js/main_html5.js

```diff
@@ -485,50 +486,64 @@ require("../../../html5-common/js/utils/environment.js");
      * @param {object} closedCaptions The closedCaptions object
      * @param {object} params The params to set with closed captions
      */
-    this.setClosedCaptions = function(language, closedCaptions, params) {
+    this.setClosedCaptions = _.bind(function(language, closedCaptions, params) {
+      //Remove and disable current captions before setting new ones.
       $(_video).find('.' + TRACK_CLASS).remove();
+      this.setClosedCaptionsMode(OO.CONSTANTS.CLOSED_CAPTIONS.DISABLED);
       if (language == null) return;
 
       var captionMode = (params && params.mode) || OO.CONSTANTS.CLOSED_CAPTIONS.SHOWING;
 
-      // The textTrack added by QuickTime will not be removed by removing track element
-      // But the textTrack that we added by adding track element will be removed by removing track element.
-      // This first check is to check for live CC
-      if (OO.isSafari && _video.textTracks.length !== 0 && language == "CC") {
-        for (var i = 0; i < _video.textTracks.length; i++) {
-          if (_video.textTracks[i].kind === "captions") {
-            _video.textTracks[i].mode = captionMode;
-            _video.textTracks[i].oncuechange = onClosedCaptionCueChange;
-          } else {
-           _video.textTracks[i].mode = OO.CONSTANTS.CLOSED_CAPTIONS.DISABLED;
+      //Add the new closed captions if they are valid.
+      var captionsFormat = "closed_captions_vtt";
+      if (closedCaptions && closedCaptions[captionsFormat]) {
+        _.each(closedCaptions[captionsFormat], function(captions, languageKey) {
```

Comment:
Because when I tell VTC about closed captions found on playing, it then sends that to the skin which overrides what the skin previously thought were the available closed captions. Unless I am mistaken.